### PR TITLE
Feature set background color for state in UIButton

### PR DESCRIPTION
**Pull request by TwizzyIndy**
This PR is referenced on #1037 .

## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 5.0.
- [x] New extensions support iOS 10.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+, or use `@available` if not.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [x] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

### CODE REVIEW

File: Sources/SwifterSwift/UIKit/UIButtonExtensions.swift [hunk 1]
```diff
@@ -179,6 +179,22 @@ public extension UIButton {
             contentEdgeInsets = UIEdgeInsets(top: 0, left: insetAmount, bottom: 0, right: insetAmount)
         }
     }
+    
+    /// SwifterSwift: Set background color for specified state.
+    /// - Parameters:
+    ///   - color: UIColor.
+    ///   - forState: set the UIControl.State for the desired color.
+    func setBackgroundColor(color: UIColor, forState: UIControl.State) {
+        self.clipsToBounds = true  // maintain corner radius
```
Comment (RomanPodymov) [inline Sources/SwifterSwift/UIKit/UIButtonExtensions.swift:188]:
```suggestion
        clipsToBounds = true  // maintain corner radius
```

**Review by RomanPodymov — CHANGES_REQUESTED** [review]

File: Sources/SwifterSwift/UIKit/UIButtonExtensions.swift [hunk 2]
```diff
@@ -179,6 +179,22 @@ public extension UIButton {
...
         }
     }
+    
+    /// SwifterSwift: Set background color for specified state.
+    /// - Parameters:
+    ///   - color: UIColor.
+    ///   - forState: set the UIControl.State for the desired color.
+    func setBackgroundColor(color: UIColor, forState: UIControl.State) {
+        clipsToBounds = true  // maintain corner radius
+        UIGraphicsBeginImageContext(CGSize(width: 1, height: 1))
```
Comment (LucianoPAlmeida) [inline Sources/SwifterSwift/UIKit/UIButtonExtensions.swift:189]: Should we use `UIGraphicsImageRenderer` APIs here? I remember a while ago that this is the recomended way to do it. See https://github.com/SwifterSwift/SwifterSwift/pull/721/files for example. I guess (not sure) we have iOS 10 as minimum for a while so conditional checking ...

Comment (TwizzyIndy) [reply Sources/SwifterSwift/UIKit/UIButtonExtensions.swift:189]: Thanks. That really makes sense. I changed using `UIGraphicsImageRenderer` for `iOS 10.0` and `tvOS 10.0`. It worked fine and tests passed.

```swift
    /// SwifterSwift: Set background color for specified state.
    /// - Parameters:
    ///   - color: UIColor.
    ///   - forState: set the UIControl.State for the desired color.
    func setBackgroundColor(color: UIColor, forState: UIControl.State) {
        clipsToBounds = true  // maintain corner radius
        
        if #available(iOS 10.0, tvOS 10.0, *) {
            let colorImage = UIGraphicsImageRenderer(size: CGSize(width: 1, height: 1)).image {
                context in
                color.setFill()
                context.fill(CGRect(x: 0, y: 0, width: 1, height: 1))
                draw(.zero)
            }
            setBackgroundImage(colorImage, for: forState)
            return
        }
        
        UIGraphicsBeginImageContext(CGSize(width: 1, height: 1))
        if let context = UIGraphicsGetCurrentContext() {
            context.setFillColor(color.cgColor)
            context.fill(CGRect(x: 0, y: 0, width: 1, height: 1))
            let colorImage = UIGraphicsGetImageFromCurrentImageContext()
            UIGraphicsEndImageContext()
            setBackgroundImage(colorImage, for: forState)
        }
    }
```

Comment (LucianoPAlmeida) [reply Sources/SwifterSwift/UIKit/UIButtonExtensions.swift:189]: Yes for sure! 
Do we need the if #available? I think we have minimums as 10 already

File: Sources/SwifterSwift/UIKit/UIButtonExtensions.swift [hunk 3]
```diff
@@ -179,6 +179,21 @@ public extension UIButton {
             contentEdgeInsets = UIEdgeInsets(top: 0, left: insetAmount, bottom: 0, right: insetAmount)
         }
     }
+    
+    /// SwifterSwift: Set background color for specified state.
+    /// - Parameters:
+    ///   - color: UIColor.
```
Comment (LucianoPAlmeida) [inline Sources/SwifterSwift/UIKit/UIButtonExtensions.swift:185]: Nit:
```suggestion
    ///   - color: The color of the image that will be set as background for the button in the given state.
```

**Review by LucianoPAlmeida — APPROVED** [review]

File: Sources/SwifterSwift/UIKit/UIButtonExtensions.swift (hunk 2 above)
Comment (TwizzyIndy) [reply Sources/SwifterSwift/UIKit/UIButtonExtensions.swift]: I guess we might need to check with #available. Even the PR you provided earlier checked with it [here](https://github.com/SwifterSwift/SwifterSwift/pull/721/files). Shall we revert it back?

> 1 Error
> 🚫	**[Sources/SwifterSwift/UIKit/UIButtonExtensions.swift#L190](https://github.com/TwizzyIndy/SwifterSwift/blob/a16139b5efcb000a524817afcb9c86f14a3b24d8/Sources/SwifterSwift/UIKit/UIButtonExtensions.swift#L190)**: 'UIGraphicsImageRenderer' is only available in tvOS 10.0 or newer
> ` let colorImage = UIGraphicsImageRenderer(size: CGSize(width: 1, height: 1)).image { context in `
> 1 Message
> 📖	Thank you for submitting a pull request to SwifterSwift. The team will review your submission as soon as possible.
> Generated by 🚫 [Danger](https://danger.systems/)

Comment (LucianoPAlmeida) [reply Sources/SwifterSwift/UIKit/UIButtonExtensions.swift]: Yeah, sorry for that... I was pretty sure we had this as mim

**Review by LucianoPAlmeida — APPROVED** [review]

**Review by RomanPodymov — APPROVED** [review]

**Comment by SD10** [comment]
Thank you for contributing to SwifterSwift! I've invited you to join the SwifterSwift GitHub organization - no pressure to accept! If you'd like more information on what that means, check out our [contributing guidelines](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md). Feel free to reach out if you have any questions! 😃

**Comment by LucianoPAlmeida** [comment]
Thanks @TwizzyIndy @RomanPodymov!

**Pull request merged**